include/cutlass/gemm/collective/intel_pvc_mma.hpp

```diff
@@ -142,74 +140,102 @@ struct CollectiveMma<
+                  "D tensor must be rmem resident.");
+    static_assert(
+        is_tuple<typename TensorA::engine_type::iterator::value_type>::value,
+        "A tensor must be a tuple iterator.");
+    static_assert(
+        is_tuple<typename TensorB::engine_type::iterator::value_type>::value,
+        "B tensor must be a tuple iterator.");
+    static_assert(is_rmem<FrgTensorC>::value,
+                  "C tensor must be rmem resident.");
 
     // Tensor to hold input data
-    Tensor tAr = make_tensor<typename TiledMma::ValTypeA>(Shape<Int<get<0>(SubgroupTileShape{}) * FragsK>, _1>{});
-    Tensor tBr = make_tensor<typename TiledMma::ValTypeB>(Shape<Int<get<1>(SubgroupTileShape{}) / 2>, Int<FragsN>>{});
+    Tensor tAr = make_tensor<typename TiledMma::ValTypeA>(
+        Shape<Int<sg_tile_m * FragsK>, Int<1>>{});
+
+    constexpr int version =
+        is_same_v<GmemTiledCopyB, XE_2D_U16x16x16x2x1_V> ? 1 : 2;
+    Tensor tBr = make_tensor<typename TiledMma::ValTypeB>(
+        Shape<Int<sg_tile_k * version>, Int<FragsN / version>>{});
 
     Tensor tAr_view = make_tensor(static_cast<decltype(tAr) &&>(tAr).data(),
-                            Shape<Int<VecA>, Int<FragsM>, Int<FragsK>>{});
+                                  Shape<Int<VecA>, Int<FragsM>, Int<FragsK>>{});
     Tensor tBr_view = make_tensor(static_cast<decltype(tBr) &&>(tBr).data(),
-                            Shape<Int<VecB>, Int<FragsN>, Int<FragsK>>{},
-                            Stride<_1, Int<get<1>(SubgroupTileShape{}) / 2>, Int<VecB>>{});
+                                  Shape<Int<VecB>, Int<FragsK>, Int<FragsN>>{});
 
     // Instantiate the M MA object
     TiledMma tiled_mma;
 
+    int K = size<1>(mainloop.gmem_tiled_copy_a.tensor);
+    Tensor tAi = make_tensor(
+        make_inttuple_iter(
+            *gA.data() +
+            make_coord((get_sub_group_id() % sg_per_wg_n % 4) * DpasM, 0)),
+        make_layout(make_shape(_1{}, _1{}, K),
+                    make_stride(_1{}, E<0>{}, E<1>{})));
+    Tensor tBi = make_tensor(
+        make_inttuple_iter(
+            *gB.data() +
+            make_coord((get_sub_group_id() / sg_per_wg_n / 2 % 2) * DpasK,
+                       (get_sub_group_id() / sg_per_wg_n % 2 * 2) * DpasN)),
+        make_layout(make_shape(_1{}, K, _1{}),
+                    make_stride(_1{}, E<0>{}, E<1>{})));
     //
     // Mainloop
     //
-   for (int k_tile = 0, k = 0; k_tile < k_tile_count; ++k_tile, k += get<2>(MmaAtomShape()) * FragsK)
-   {
-     // Copy gmem to rmem for the first k_tile
-     copy(mainloop.gmem_tiled_copy_a, gA(_,_,k), tAr);
-     copy(mainloop.gmem_tiled_copy_b, gB(_,_,k/2), tBr);
-
-     cute::gemm(tiled_mma, accum, tAr_view, tBr_view, src_accum);
-   }
+    int prefetch_k = 0;
+    for (int i = 0; i < 3; i++) {
+      prefetch(mainloop.gmem_tiled_copy_a, tAi(_, _, prefetch_k));
+      prefetch(mainloop.gmem_tiled_copy_b, tBi(_, prefetch_k, _));
+      prefetch_k += sg_tile_k;
+    }
```

Comment:
I think we should avoid not naming constants and directly putting numeric values.
Also I think it is more confusing to have seperate loop variable rather than directly looping over `prefetch_k`. But that is a matter of taste (up to you whether you want to change it).
```suggestion
    constexpr int kPrefetch_k_dist = 3;
    for (; prefetch_k < kPrefetch_k_dist*sg_tile_k ; prefetch_k += sg_tile_k) {
      prefetch(mainloop.gmem_tiled_copy_a, tAi(_, _, prefetch_k));
      prefetch(mainloop.gmem_tiled_copy_b, tBi(_, prefetch_k, _));
    }
```